samples/js/dropbox/slurp/src/main.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import argv from 'yargs';
+import {
+  default as fetch,
+  Request,
+  Headers,
+} from 'node-fetch';
+import {
+  DatasetSpec,
+  jsonToNoms,
+  newStruct,
+} from '@attic/noms';
+import {stripIndent} from 'common-tags';
+
+const apiHost = 'https://api.dropboxapi.com/2/';
+const clearLine = '\x1b[2K\r';
+
+const args = argv
+  .usage(stripIndent`
+    Downloads metadata from the Dropbox API
+
+    Usage: node . --access-token=<token> <dest-dataset>`)
+  .demand(1)
+  .option('access-token', {
+    describe: 'Dropbox oauth access token',
+    type: 'string',
+    demand: true,
+  })
+  .argv;
+
+main().catch(ex => {
+  console.error(ex);
+  process.exit(1);
+});
+
+async function main(): Promise<void> {
+  const outSpec = DatasetSpec.parse(args._[0]);
+  if (!outSpec) {
+    throw 'invalid destination dataset spec';
+  }
+
+  const [db, out] = outSpec.dataset();
+
+  const data = [];
+  let count = 0;
+  let resp = await callDropbox(apiHost + 'files/list_folder', {
+    path: '',
+    recursive: true,
+    'include_media_info': true,
+  });
+  while (resp) {
+    count += resp.entries.length;
+    process.stdout.write(clearLine + `Slurped ${count} items...`);
+    data.push(resp);
+    if (resp.has_more) {
+      resp = await callDropbox(apiHost + 'files/list_folder/continue', {
+        cursor: resp.cursor,
+      });
+    } else {
+      resp = null;
+    }
+  }
+
+  await db.commit(out, jsonToNoms(data), {
+    meta: newStruct('', {
+      date: new Date().toISOString(),
+    }),
+  });
+  await db.close();
+  process.stdout.write(clearLine);
+}
+
+async function callDropbox(url: string, body: ?any = null): Promise<any> {
+  const params = {
+    method: 'POST',
+    headers: new Headers({
```

Comment:
At least in the web version of fetch you do not need to convert the object to a Headers object.